packages/signers/src/message-modifying-signer.ts

```diff
@@ -0,0 +1,30 @@
+import { Address, isAddress } from '@solana/addresses';
+
+import { SignableMessage } from './signable-message';
+
+/** Defines a signer capable of signing messages. */
+export type MessageModifyingSigner<TAddress extends string = string> = Readonly<{
+    address: Address<TAddress>;
+    modifyAndSignMessage(messages: readonly SignableMessage[]): Promise<readonly SignableMessage[]>;
```

Comment:
Would it make sense to return the new `SignableMessage` _and_ the new `SignatureDictionary`? Since the dictionary will now have one entry.

Reply:
I'm not sure I understand the benefit of that since the the `SignableMessage` already contains a `SignatureDictionary`?

Reply:
Oops my bad, that's cool then 😅